smartsim/entity/ensemble.py

```diff
@@ -298,3 +307,151 @@ def _read_model_parameters(self):
                     + "Must be list, int, or string."
                 )
         return param_names, parameters
+
+
+    def add_ml_model(self,
+                     name,
+                     backend,
+                     model=None,
+                     model_path=None,
+                     device="CPU",
+                     devices_per_node=1,
+                     batch_size=0,
+                     min_batch_size=0,
+                     tag="",
+                     inputs=None,
+                     outputs=None):
+        """A TF, TF-lite, PT, or ONNX model to load into the DB at runtime
+
+        Each ML Model added will be loaded into an
+        orchestrator (converged or not) prior to the execution
+        of every entity belonging to this ensemble
+
+        One of either model (in memory representation) or model_path (file)
+        must be provided
+
+        :param name: key to store model under
+        :type name: str
+        :param model: model in memory
+        :type model: str, optional # TODO figure out what to type hint this as
```

Comment:
Could you explain what you mean by this?

Reply:
I think this was a leftover from old code (which was copied and pasted). In some cases this should be a string, in other cases a byte string, I think. Suggestions on how to express this?

Reply:
I know if I where to do this with the typing package it would look something like `Optional[Union[str, bytes]]` or with 3.10 syntax `str | bytes | None`. Perhaps `str, bytes, optional` or directly use the 3.10 syntax?

Reply:
Let's go for 3.10 syntax!